rfc-087-use-libraries-where-possible.md

```diff
@@ -0,0 +1,98 @@
+# Use Existing Libraries Where Possible
+
+## Summary
+
+We should use existing open source libraries wherever possible,
+rather than writing our own code, but we should ensure that those
+libraries are trustworthy as best we can.
+
+## Problem
+
+Writing and maintaining code takes time, and any code is a possible
+source of bugs. When we implement our own solutions for problems
+that exist elsewhere, we may repeat work, increase development time,
+and increase the potential for error.
+
+## Proposal
+
+When we need to add new generic functionality to our code, we have a
+choice of whether to write our own implementation, or use an
+off-the-shelf open source library.
+
+By default, we SHOULD aim to find and reuse existing open source
+libraries that fulfil the requirement rather than writing our own.
+
+Using existing code is preferable in order to get better tested
+code, a lower maintenance burden, and more features than we might
+normally implement ourselves.
+
+Some examples might include:
+
+- Date handling and formatting
+- Language inflection
+- Queueing and background job running
+- Connecting to remote APIs
+
+The need for this will differ across languages and frameworks. For
+instance, Rails provides many capabilities that separate libraries
+would be needed for when working in Node.js. Conversely, frontend
+code may wish to avoid dependencies to keep file sizes low.
+
+Acquiring dependencies is not free, however. Developers MUST check
+them for license compatibility, and SHOULD assess them for quality
+and reliability, so that we are sure they will be a net positive
+addition to our projects.
```

Comment:
Can we / should we keep a repository of packages we've reviewed (including version and date checked), to minimize the repeated work in doing these checks. We could then think about building a GitHub action to confirm that all direct dependencies have been checked, and recently / for a semver compatible version if that's useful.

Reply:
Adding this as a suggestion in the consequences section